### PR TITLE
[1516] Create product only when all required fields are valid

### PR DESCRIPTION
**Pull request by jwm0**
Since product creation is split into multiple mutations and some of the required fields are scattered throughout them, I propose to only create the product when validation of all of them is successful. Due to mutations needing to be fired in order we have to first create a product to fire up the rest of them, which always results in a product being created and therefore the form state was lost. To avoid a larger refactor we remove the previously created product if the subsequent mutations fail, this way we don't duplicate products and the UI requirements are matched with the inner logic.

https://app.clickup.com/t/2549495/SALEOR-1516
**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1516 [Create Product] Values are missing after validation error](https://app.clickup.com/t/2549495/SALEOR-1516)

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
I believe that all this logic should be contained in product create handler rather than in component

File: src/products/views/ProductCreate/ProductCreate.tsx [hunk 1]
```diff
@@ -133,17 +133,32 @@ export const ProductCreateView: React.FC = () => {
...
+    }
+  };
+
+  React.useEffect(() => {
+    const productId = productCreateOpts.data?.productCreate?.product?.id;
+
+    if (productCreateComplete && productId) {
+      handleSuccess(productId);
+    }
+  }, [productCreateComplete]);
```
Comment (dominik-zeglen) [inline src/products/views/ProductCreate/ProductCreate.tsx:161]: I think we can add this to handler too and skip `productCreateComplete` state from component 🤷‍♂️

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Comment by mlamperski** [comment]
Tested. Issue doesn't occure on deployed environment.

**Review by orzechdev — APPROVED** [review]

**Pull request merged**